packages/nimble-components/src/list-option-group/index.ts

```diff
@@ -0,0 +1,121 @@
+import { DesignSystem, FoundationElement } from '@microsoft/fast-foundation';
+import { observable, attr, volatile } from '@microsoft/fast-element';
+import { styles } from './styles';
+import { template } from './template';
+import { ListOption } from '../list-option';
+
+declare global {
+    interface HTMLElementTagNameMap {
+        'nimble-list-option-group': ListOptionGroup;
+    }
+}
+
+/**
+ * A nimble-styled HTML listbox option
+ */
+export class ListOptionGroup extends FoundationElement {
+    /**
+     * The label for the group.
+     *
+     * @public
+     * @remarks
+     * If a label is also provided via slotted content, the label attribute
+     * will have precedence.
+     */
+    @attr
+    public label?: string;
+
+    /**
+     * The hidden state of the element.
+     *
+     * @public
+     * @defaultValue - false
+     * @remarks
+     * HTML Attribute: hidden
+     */
+    @attr({ mode: 'boolean' })
+    public override hidden = false;
+
+    /**
+     * @internal
+     * This attribute is required to allow use-cases that offer dynamic filtering
+     * (like the Select) to visually hide groups that are filtered out, but still
+     * allow users to use the native 'hidden' attribute without it being affected
+     * by the filtering process.
+     */
+    @attr({ attribute: 'visually-hidden', mode: 'boolean' })
```

Comment:
Talked offline but I'm generally concerned about how much internal state is on list-option-group and list-option.
The concern is around option and group element recycling.
Some thoughts:
- Should we clear internal state on connect / disconnect (option or group re-used between controls or re-ordered)
- Should we clear internal state for any public attribute change (value, selected, hidden)? (options are updated in-place to represent new values, i.e. in manual filtering we only populate the first 5 options and as the filter changes the state of the options change but they are not attached / detached / re-ordered)
- Is WeakMap a useful tool for managing state of the options in the select itself instead of on the option / group instances? (may not be since we want to leverage internal state as attributes for styling)
- Is [sibling combinators](https://developer.mozilla.org/en-US/docs/Web/CSS/Next-sibling_combinator) a useful tool for a different approach to state management? i.e. groups just reflect an attribute that says they are empty and use sibling combinators to declaratively style bottom / top separator instead of relying on imperative "show" configuration

Reply:
So I've done a re-factor that may allay some of the concerns presented here (but perhaps not).

The main thrust of the re-factor is removing the management of the separator visual states from the `filterOptions` implementation in order to support scenarios like manual filtering with groups. This is now controlled by responding to `hidden` and `visuallyHidden` state changes on the groups and options themselves. The `visuallyHidden` state of the group is now managed by the group itself, and not within the `filterOptions` implementation, again to support the manual filtering with groups case.

Sibling combinators do not seem possible to manage the necessary visual states as there is no CSS syntax to handle skipping over hidden groups.

So my summary of responses to the bullet points are (and I'm certain there could be more discussion to be had):
- I don't think this is necessary at the moment without identifying a use-case that won't function properly.
- Same as above
- I don't believe this is necessary due to how the `visuallyHidden` state gets reset when needed already
- Sibling combinators won't work because we do rely on styling to use the internal `visuallyHidden` state.

Unless we think the current implementation within this PR is problematic for known use-cases, I propose we move forward with it, and if there is an identified scenario outside of known use-cases that is problematic, I think this should be tracked by a Tech Debt issue.

@jattasNI @rajsite @mollykreis